### DISCUSSION TITLE
ScreenGridLayer + DataFilterExtension

### DISCUSSION THREAD

**Discussion by hyundotio** [visgl/deck.gl#5467] [Q&A]
Hello everyone...

Got a question on ScreenGridLayer + DataFilterExtension…

I see in the DataFilterExtension docs that SGL is supported long as GPU aggregation is being done (which in my instance, is). However, data filtering doesn’t seem to work.

BTW, I tried this using a ScatterPlotLayer and filtering does work with the same properties.

Anyone know of a good example how to use ScreenGridLayer & DataFilterExtension together or know why it isn’t working?

```
const dataFilter = new DataFilterExtension({
  filter:1,
  countItems: true
});

const screenGridOptions = {
  cellSizePixels: 16,
  cellMarginPixels: 1
}

new ScreenGridLayer({
        data,
        id: 'grid',
        getPosition: d => [d.longitude, d.latitude],
        cellMarginPixels: screenGridOptions.cellMarginPixels,
        cellSizePixels: screenGridOptions.cellSizePixels,
        extensions: [dataFilter],
        getFilterValue: d => d.timestamp,
        filterRange: [filterValue[0], filterValue[1]],
        pickable: true,
      })
```

Also when I try  ```onFilteredItemsChange: count => console.log(count)``` it returns  0 as well :confused:

Thank you for your help!

**Comment by sclebal** [comment]
Hi @hyundotio, I don't see anything weird in your code. Can you provide an example?

**Reply by sclebal** [reply]
Btw, check if your browser supports WebGL2:

```
import {isWebGL2} from 'luma.gl';
function onWebGLInitialized(gl) {
  const webGL2Supported = isWebGL2(gl);
  console.log(webGL2Supported);
}
<DeckGL
  ...
  onWebGLInitialized={onWebGLInitialized}
/>
```

GPU Aggregation is only available in WebGL2
